packages/js/msgpack/src/index.ts

```diff
@@ -22,14 +16,19 @@ extensionCodec.register({
       for (const [key, value] of object) {
         optimized[key] = value;
       }
-      return encode(optimized);
+      return msgpackEncode(optimized);
```

Comment:
So now you're calling `msgpackEncode` instead of calling this function recursively, right? Seems simpler than what you thought at first :P

Reply:
It just worked out haha